test/ToastNotificationTests/APITests.cpp

```diff
@@ -583,5 +583,50 @@ namespace Test::ToastNotifications
         {
             RunTestUnpackaged(L"VerifyIconPathExists_Unpackaged", testWaitTime());
         }
+
+        TEST_METHOD(VerifyRegisterWithNullDisplayNameFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithNullDisplayNameFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithNullIconFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithNullIconFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithNullDisplayNameAndNullIconFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithNullDisplayNameAndNullIconFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithDisplayNameAndIcon_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithDisplayNameAndIcon_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithDisplayNameAndInvalidIconPathFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithDisplayNameAndInvalidIconPathFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithEmptyDisplayNameFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithEmptyDisplayNameFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithEmptyIconFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithEmptyIconFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithEmptyDisplayNameAndEmptyIconFail_Unpackaged)
+        {
+            RunTestUnpackaged(L"VerifyRegisterWithEmptyDisplayNameAndEmptyIconFail_Unpackaged", testWaitTime());
+        }
+
+        TEST_METHOD(VerifyRegisterWithAssetsFail)
+        {
```

Comment:
I assume that we run a single test for packaged just to confirm that register works for packaged, and that we rely on the more complete suite of unpackaged tests to actually verify Register()'s actual behavior.

It would be nice if, we could just run the suite twice: once for unpackaged and one for packaged. It seems the only difference is whether RunTestUnpackaged or RunTest is used to launch the test. It feels like dependency injection could help here. 

Not suggesting you do it now, that's probably beyond the point of this PR but that could be a nice future refactor.

Reply:
It's possible to detect at compile time if you're built with WindowsAppSDKSelfContained=true or false. At least, I think it is now in Foundation with recent changes I made to .props/.targets

Our test infrastructure does tend to build everything both ways though it was only running SelfContained=false. My PR in the Aggregator fixes that (going in soon).

Sorry limited network access at the moment. If you grip for SelfContained across build\* you should find the C++ and C# symbols code can check for.
Though I suggest not messing with that until my changes go in as that has significant updates to the plumbing